contracts/contracts/strategies/GeneralizedUniswapV3Strategy.sol

```diff
@@ -0,0 +1,1144 @@
+        //     "Invalid strategy for asset"
+        // );
+
+        poolTokens[_asset].reserveStrategy = _reserveStrategy;
+
+        emit ReserveStrategyChanged(_asset, _reserveStrategy);
+    }
+
+    /**
+     * @notice Get reserve strategy of the given asset
+     * @param _asset Address of the asset
+     * @return reserveStrategyAddr Reserve strategy address
+     */
+    function reserveStrategy(address _asset)
+        external
+        view
+        onlyPoolTokens(_asset)
+        returns (address reserveStrategyAddr)
+    {
+        return poolTokens[_asset].reserveStrategy;
+    }
+
+    /**
+     * @notice Change the minimum deposit threshold for the supported asset
+     * @param _asset Asset to set the threshold
+     * @param _minThreshold The new deposit threshold value
+     */
+    function setMinDepositThreshold(address _asset, uint256 _minThreshold)
+        external
+        onlyGovernorOrStrategist
+        onlyPoolTokens(_asset)
+    {
+        PoolToken storage token = poolTokens[_asset];
+        token.minDepositThreshold = _minThreshold;
+        emit MinDepositThresholdChanged(_asset, _minThreshold);
+    }
+
+    function setSwapsPaused(bool _paused) external onlyGovernorOrStrategist {
+        swapsPaused = _paused;
+        emit SwapsPauseStatusChanged(_paused);
+    }
+
+    function setMaxSwapSlippage(uint24 _maxSlippage)
+        external
+        onlyGovernorOrStrategist
+    {
+        maxSwapSlippage = _maxSlippage;
+        // emit SwapsPauseStatusChanged(_paused);
+    }
+
+    // function setMinSwapPrice(address _asset, uint256 _price) external onlyGovernorOrStrategist {
+    //     maxSwapSlippage = _maxSlippage;
+    //     // emit SwapsPauseStatusChanged(_paused);
+    // }
+
+    /***************************************
+            Deposit/Withdraw
+    ****************************************/
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        onlyPoolTokens(_asset)
+        nonReentrant
+    {
+        if (_amount > poolTokens[_asset].minDepositThreshold) {
+            IVault(vaultAddress).depositForUniswapV3(_asset, _amount);
+            // Not emitting Deposit event since the Reserve strategy would do so
+        }
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function depositAll() external override onlyVault nonReentrant {
+        _depositAll();
+    }
+
+    /**
+     * @notice Deposits all undeployed balances of the contract to the reserve strategies
+     */
+    function _depositAll() internal {
+        uint256 token0Bal = IERC20(token0).balanceOf(address(this));
+        uint256 token1Bal = IERC20(token1).balanceOf(address(this));
+        if (
+            token0Bal > 0 && token0Bal >= poolTokens[token0].minDepositThreshold
+        ) {
+            IVault(vaultAddress).depositForUniswapV3(token0, token0Bal);
```

Comment:
Maybe we could rename this to `depositToUniswapV3ReserveStrategy`.

I keep forgetting what this function does and need to check it out

Reply:
Noted. Will change it